core/silkworm/types/block.cpp

```diff
@@ -34,6 +34,20 @@ bool operator==(const BlockBody& a, const BlockBody& b) {
     return a.transactions == b.transactions && a.ommers == b.ommers;
 }
 
+void Block::recover_senders(const ChainConfig& config) {
+    uint64_t block_number{header.number};
+    bool homestead{config.has_homestead(block_number)};
+    bool spurious_dragon{config.has_spurious_dragon(block_number)};
+
+    for (Transaction& txn : transactions) {
+        if (spurious_dragon) {
+            txn.recover_sender(homestead, config.chain_id);
```

Comment:
There's no trapping for errors in txn.recover_senders so we don't know if it succeeds.

Question ... as we already have senders recovered and persisted on db why don't we get them from there ?

Reply:
Nevermind. Disregard it.
This is wasm ... no database

Reply:
If senders are not recovered, block execution will fail with `kMissingSender`.